docs/howto/send-notifications.rst

```diff
@@ -0,0 +1,189 @@
+Send Notifications
+==================
+
+Taskcluster can `send notifications`_ when a task transitions states.
+
+Notification Types
+------------------
+
+There are four supported notification types:
+
+1. Email - An email can be sent to any address.
+2. Pulse - A `pulse event`_ can be triggered.
+3. Matrix - A Matrix room can be pinged. Requires the Taskcluster deployment to
+   be configured with Matrix credentials and be invited to the room.
+4. Slack - A slack channel can be pinged. Requires the Taskcluster deployment to
+   be configured with Slack credentials and be invited to the room.
+
+.. _status-types:
+
+Status Types
+------------
+
+A notification can be sent on the following task status changes:
+
+- ``on-defined`` - When the task is first submitted to Taskcluster.
+- ``on-pending`` - When the task transitions to the ``pending`` status.
+- ``on-running`` - When the task transitions to the ``running`` status.
+- ``on-completed`` - When the task finishes successfully.
+- ``on-failed`` - When the task finishes in the ``failed`` state.
+- ``on-exception`` - When the task finishes in the ``exception`` state.
+- ``on-resolved`` - When the task finishes in any state.
+- ``on-transition`` - When any of the above are triggered.
+
+Adding Notifications to a Task
+------------------------------
+
+Taskgraph supports these notifications via the ``notify`` transforms. Once these
+have been added, a *notify* section can be used:
+
+.. code-block:: yaml
+
+   transforms:
+       - taskgraph.transforms.notify:transforms
+
+   tasks:
+     my-task:
+       # <rest of task definition>
+       notify:
+         recipients:
+           - type: email
+             address: list@example.com
+             status-type: on-resolved
+           - type: slack-channel
+             channel-id: KQP51F7GF
+             status-type: on-failed
+
+The above task will notify the ``list@example.com`` mailing list anytime the
+task resolves for any reason, as well as ping the specified Slack channel
+whenever the task fails. A Matrix room can be notified using ``type: matrix``
+and ``room-id: <id>``, and a pulse route can be notified using ``type: pulse``
+and ``routing-key: <key>``.
+
+For each recipient, the ``status-type`` key can be any one of the
+aforementioned :ref:`status types <status-types>`. If not specified, it
+defaults to ``on-completed``.
+
+.. note::
+
+   For a full reference, see the :py:data:`notify schema <taskgraph.transforms.notify.NOTIFY_SCHEMA>`.
+
+Interpolating Recipients
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+Each recipient value can be interpolated using Python's `format string syntax`_
+and the ``task`` and :class:`kind config <taskgraph.generator.Kind>` objects as
+context. For example:
+
+.. code-block:: yaml
+
+   notify:
+     recipients:
+       - type: email
+         address: "{task[name]}-subscribers@example.com"
```

Comment:
Interesting! Do you know if any team has ever had the need for including the task name in the recipient?

Reply:
Not `task[name]` specifially, but Gecko does interpolate this value in a few places, e.g:
https://searchfox.org/mozilla-central/source/taskcluster/ci/release-notify-push/kind.yml#29